controllers/syncer/call_home.go

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package syncer
+
+import (
+	"github.com/imdario/mergo"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/IBM/ibm-block-csi-operator/controllers/internal/ibmblockcsi"
+	"github.com/IBM/ibm-block-csi-operator/pkg/config"
+	"github.com/IBM/ibm-block-csi-operator/pkg/util/boolptr"
+	"github.com/presslabs/controller-util/mergo/transformers"
+	"github.com/presslabs/controller-util/syncer"
+)
+
+const (
+	callHomeContainerName = "ibm-block-csi-call-home"
+	secretVolumeName      = "call-home-secret-dir"
+	cronSchedule          = "0 0 * *"
+)
+
+type callHomeSyncer struct {
+	driver *ibmblockcsi.IBMBlockCSI
+	obj    runtime.Object
+}
+
+// NewCallHomeSyncer returns a syncer for call home
+func NewCallHomeSyncer(c client.Client, scheme *runtime.Scheme, driver *ibmblockcsi.IBMBlockCSI) syncer.Interface {
+	obj := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        config.GetNameForResource(config.CallHome, driver.Name),
+			Namespace:   driver.Namespace,
+			Annotations: driver.GetAnnotations("", ""),
+			Labels:      driver.GetLabels(),
+		},
+	}
+
+	sync := &callHomeSyncer{
+		driver: driver,
+		obj:    obj,
+	}
+
+	return syncer.NewObjectSyncer(config.CallHome.String(), driver.Unwrap(), obj, c, func() error {
+		return sync.SyncFn()
+	})
+}
+
+func (s *callHomeSyncer) SyncFn() error {
+	out := s.obj.(*batchv1.CronJob)
+
+	//Run once a day at midnight
```

Comment:
this is easier to maintain next to the actual value, and with a space after `//`

Reply:
done

Reply:
by actual value I mean where the string literal is

Reply:
done